scio-core/src/main/scala/com/spotify/scio/io/dynamic/syntax/SCollecionSyntax.scala

```diff
@@ -122,12 +122,10 @@ final class DynamicGenericRecordSCollectionOps[T <: GenericRecord](private val s
       val nm = new JHashMap[String, AnyRef]()
       nm.putAll(metadata.asJava)
       val sink = beam.AvroIO
-        .sinkViaGenericRecords(schema, new RecordFormatter[T] {
-          override def formatRecord(element: T, schema: Schema): GenericRecord =
-            element
-        })
+        .sink[T](schema)
```

Comment:
hm, I think this will use `ReflectDatumWriter` instead of `GenericDatumWriter` which is preferred in this case 😢 

Same thing in the case below.

Reply:
Good catch. Boy is AvroIO tricky!

Reply:
So… beam has `AvroIO.Sink`which extends `FileIO.Sink`  and `AvroSink` which extends `FileBasedSink`. `AvroIO.Sink` and `AvroSink` are completelly urealated classes. (That’s not confusing...). `AvroIO.Sink` is implemented using either `RecordFormatter` which is deprecated, or `ReflectDatumWriter` (great…). So **`AvroIO.Sink` is to be avoided.**

`AvroIO` has `AvroOI.Write` and `AvroOI.TypedWrite`.  `AvroOI.Write` is delegating to `AvroOI.TypedWrite` and only exists for compatibility reasons.

`AvroOI.TypedWrite` is implemented using `AvroSink` and does not rely on deprecated APIs or reflection (:tada:) **BUT** it does not support dynamic destination or rather it does **BUT** does not expose the API... 

`AvroIO.Write` does expose `public Write<T> to(DynamicAvroDestinations<T, ?, T> dynamicDestinations)` but `DynamicAvroDestinations` is abstract and does not have a public implementation (other than `constantDestinations`) .

I feel like Beam is trolling me.